src/indexes/createIndexes.ts

```diff
@@ -46,27 +46,31 @@ export function createIndexes<T extends IndexableObj>(
     Deleter<T>
   >(isNotUndefined);
 
-  const captureObject: Captor<T> = (obj) => {
-    const proxy = new Proxy(
-      Object.assign(obj, {
-        deleteFromIndex() {
-          deleters.forEach((deleteFromIndexFunc) => deleteFromIndexFunc(proxy));
-        },
-      }),
-      {
-        set(target: T, propName: never, newValue: any): boolean {
-          updaters.forEach((updateIndexFunc) =>
-            updateIndexFunc(proxy, propName, newValue)
-          );
-          // @ts-ignore
-          target[propName] = newValue;
-          return true;
-        },
-      }
-    ) as T & Extension;
-    ingestors.forEach((ingestIntoIndex) => ingestIntoIndex(proxy));
+  const captureObject: Captor<T> = (obj: T): IndexedObj<T> => {
+    const proxy = new Proxy(obj, {
+      set(target: T, propName: never, newValue: any): boolean {
+        updaters.forEach((updateIndexFunc) =>
+          updateIndexFunc(proxy, propName, newValue)
+        );
+        // @ts-ignore
+        target[propName] = newValue;
+        return true;
+      },
+    });
 
-    return proxy;
+    const extendedObject = Object.assign(proxy, {
+      deleteFromIndex() {
+        deleters.forEach((deleteFromIndexFunc) => deleteFromIndexFunc(proxy));
+      },
+      getTarget() {
+        return obj;
+      },
+      isProxy: true as const,
```

Comment:
With these two values, we could easily fork a deep clone util like https://github.com/lukeed/klona with a couple lines of code to check if the object is a proxy, and if so, pass the target object into the deep clone function call 🤔